### PR TITLE
[msbuild] Copy all the files in a framework to the Mac for the FilterStaticFrameworks task. Fixes #13804.

### LINKED ISSUE #13804: Remote-deploy frameworks ends up in the wrong cache folder
### Steps to Reproduce

1. Create new project `dotnet new maui`
2. Add package `dotnet add package Esri.ArcGISRuntime.runtimes.iOS -v 100.13.0`
3. Open Solution in VS2022 17.1 preview 3
4. Connect to a remote build host
5. Build/deploy to a simulator

### Expected Behavior
Application builds and runs

### Actual Behavior
Build error:
```
1>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.2.200-preview.12.4\targets\Xamarin.Shared.Sdk.targets(616,3): error MT7091: File 'C:/Users/mn/.nuget/packages/esri.arcgisruntime.runtimes.ios/100.13.0/buildTransitive/net6.0-ios13/../../framework/iossimulator-x64/native/Runtimecore.framework/Runtimecore' is not a valid framework: Could not find a part of the path "/Users/mn/Library/Caches/Xamarin/mtbs/builds/mauip12/cac205ab3281b0a94658323c40bdaf2c56e8cd0f4e6afd1fd20dcdc9e62a7914/C:/Users/mn/.nuget/packages/esri.arcgisruntime.runtimes.ios/100.13.0/framework/iossimulator-x64/native/Runtimecore.framework/Runtimecore".
1>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.2.200-preview.12.4\targets\Xamarin.Shared.Sdk.targets(616,3): error MT7091: File 'C:/Users/mn/.nuget/packages/esri.arcgisruntime.runtimes.ios/100.13.0/buildTransitive/net6.0-ios13/../../framework/iossimulator-x64/native/ArcGIS-simx64.framework/ArcGIS-simx64' is not a valid framework: Could not find a part of the path "/Users/mn/Library/Caches/Xamarin/mtbs/builds/mauip12/cac205ab3281b0a94658323c40bdaf2c56e8cd0f4e6afd1fd20dcdc9e62a7914/C:/Users/mn/.nuget/packages/esri.arcgisruntime.runtimes.ios/100.13.0/framework/iossimulator-x64/native/ArcGIS-simx64.framework/ArcGIS-simx64".
1
```
It seems like the full local Windows path is getting into the MacOS cache path.

### Environment

```
Microsoft Visual Studio Enterprise 2022
Version 17.1.0 Preview 3.0
VisualStudio.17.Preview/17.1.0-pre.3.0+32113.165
Microsoft .NET Framework
Version 4.8.04161

Installed Version: Enterprise

Visual C++ 2022   00476-80000-00000-AA974
Microsoft Visual C++ 2022

ASP.NET and Web Tools 2019   17.1.348.51711
ASP.NET and Web Tools 2019

Azure App Service Tools v3.0.0   17.1.348.51711
Azure App Service Tools v3.0.0

C# Tools   4.1.0-3.22062.11+cc0d213b15fdda217ad888d269d19754fa556eb4
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   17.1.10 (65f98f5)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   6.1.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Project System Tools   1.0
Tools for working with C#, VisualBasic, and F# projects.

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

TypeScript Tools   17.0.1229.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.1.0-3.22062.11+cc0d213b15fdda217ad888d269d19754fa556eb4
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.1.0-beta.21610.4+07b5673e4f2fa7630e78abe37f16b372353a7242
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Foo   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   17.1.0.296 (d17-1@b4d15b4)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   17.1.0.87 (remotes/origin/main@d50fb5635)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   17.1.8 (590388a)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   12.2.0.2 (d17-1/e9c7503)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: a5d1934
    Java.Interop: xamarin/java.interop/d17-1@7ac33610
    ProGuard: Guardsquare/proguard/v7.0.1@912d149
    SQLite: xamarin/sqlite/3.37.1@0893b3b
    Xamarin.Android Tools: xamarin/xamarin-android-tools/main@db125a7


Xamarin.iOS and Xamarin.Mac SDK   15.6.0.2 (488b7df8d)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

### Build Logs
Bin log:
[mauip12_Debug_AnyCPU_Build_2022-01-19T23_00_52.5714160-08_00.zip](https://github.com/xamarin/xamarin-macios/files/7903010/mauip12_Debug_AnyCPU_Build_2022-01-19T23_00_52.5714160-08_00.zip)

### PR DESCRIPTION
**Pull request by rolfbjarne**
The FilterStaticFrameworks task takes a list of frameworks to process, but the
files actually needed by the task are inside those frameworks, which means we
have to tell XVS that those have to be copied to the Mac for the build as well.

Fixes https://github.com/xamarin/xamarin-macios/issues/13804.

### CODE REVIEW

File: msbuild/Xamarin.MacDev.Tasks/Tasks/FilterStaticFrameworks.cs [hunk 1]
```diff
@@ -18,6 +24,19 @@ public override bool Execute ()
...
-		public IEnumerable<ITaskItem> GetAdditionalItemsToBeCopied () => Enumerable.Empty<ITaskItem> ();
+		public IEnumerable<ITaskItem> GetAdditionalItemsToBeCopied ()
+		{
+			if (FrameworkToPublish is not null) {
+				foreach (var item in FrameworkToPublish) {
+					var fw = item.ItemSpec;
+					// Copy all the files from the framework to the mac (copying only the executable won't work if it's just a symlink to elsewhere)
+					if (File.Exists (fw))
+						fw = Path.GetDirectoryName (fw);
+					foreach (var file in Directory.EnumerateFiles (fw, "*.*", SearchOption.AllDirectories)) {
```
Comment (chamons) [inline msbuild/Xamarin.MacDev.Tasks/Tasks/FilterStaticFrameworks.cs:35]: Spacing wrong here?

**Review by chamons — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Comment by dotMorten** [comment]
Could this potentially also address #13665 ?

**Review by emaf — APPROVED** [review]
I've tried these changes from VS and the build works now.

@rolfbjarne this is the build output in case you want to double check the right files are being copied https://gist.github.com/emaf/7111b787e3a5ebd93032a45d81b65481. There are 2 warnings that don't seem to be related to this, but the codesign task is having some issues to create output files on Windows.

File: msbuild/Xamarin.MacDev.Tasks/Tasks/FilterStaticFrameworks.cs (hunk 1 above)
Comment (dotMorten) [reply msbuild/Xamarin.MacDev.Tasks/Tasks/FilterStaticFrameworks.cs:35]: Space after `EnumerateFiles` ?

**Comment by rolfbjarne** [comment]
Red tests are unrelated (https://github.com/xamarin/maccore/issues/581).

**Comment by rolfbjarne** [comment]
@emaf thanks for reviewing and testing!

I think the warnings occur when trying to create a directory that already exists because an earlier output item was a file inside that directory.

**Comment by rolfbjarne** [comment]
> Could this potentially also address #13665 ?

No, I don't think so, but #13591 might have fixed it.

**Comment by dotMorten** [comment]
Thank you for the quick turnaround!

**Pull request merged**